src/controllers/StudentController.js

```diff
@@ -0,0 +1,176 @@
+import { z } from 'zod';
+import { fromZodError } from 'zod-validation-error';
+
+import { EntryExists } from '../errors/EntryExists.js';
+
+import StudentService from '../services/StudentService.js';
+
+async function list(request, response) {
+  if (request.role == 'STUDENT') {
+    return response.sendStatus(403);
+  }
+
+  try {
+    const students = await StudentService.findAll();
+    return response.json(students);
+  } catch (error) {
+    return response.status(500).json({ error: error });
+  }
+}
+
+async function add(request, response) {
+  if (request.role != 'ADMIN') {
```

Comment:
No JavaScript é mais seguro (e é padrão da comunidade) utilizar `===` e `!==` ao invés de `==` ou `!=` para evitar a coersão de tipos.

```suggestion
  if (request.role !== 'ADMIN') {
```